src/CSharp App/services/CacheService.cs

```
@@ -117,24 +122,52 @@ public void Initialize()
         }
         catch (Exception e)
         {
+            Logger.Exception(e, "Failed to initialize cache service!");
         }
     }
+
+    /// <summary>
+    /// Disposes of the cache service
+    /// </summary>
+    /// <returns></returns>
+    public Task Dispose()
+    {
+        return Task.Run(() =>
+        {
+            if (!_isInitialized) return;
+            _isInitialized = false;
+            IsProcessing = false;
+            Task.Delay(BatchDelay * 10).Wait();
+            _env.Dispose();
+        });
+    }
     
-    
+    /// <summary>
+    /// Starts listening to read requests
+    /// </summary>
+    /// <exception cref="Exception">Cache service is not initialized</exception>
     public void StartListening()
     {
+        if (!_isInitialized) throw new Exception("Cache service must be initialized before calling StartListening.");
```

should silently return early if is already listening for write requests